docs/topics/compatibility-guide-18.md

```diff
@@ -0,0 +1,623 @@
+>
+> **Deprecation cycle**:
+>
+> - 1.7.0: report a warning (or an error in the progressive mode) when a function without explicitly specified return type is
+>   called recursively in that function's body in a compound assignment operator argument
+> - 1.9.0: raise the warning to an error
+
+### Prohibit unsound calls with expected `@NotNull T` and given Kotlin generic parameter with nullable bound
+
+> **Issue**: [KT-36770](https://youtrack.jetbrains.com/issue/KT-36770)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin will prohibit method calls where a value of a potentially nullable generic type is passed
+> for a `@NotNull`-annotated parameter of a Java method
+>
+> **Deprecation cycle**:
+>
+> - 1.5.20: report a warning when an unconstrained generic type parameter is passed where a non-nullable type is expected
+> - 1.9.0: report a type mismatch error instead of the warning above,  
+>   `-XXLanguage:-ProhibitUsingNullableTypeParameterAgainstNotNullAnnotated` can be used to temporarily revert to the pre-1.8 behavior
+
+### Prohibit access to members of a companion of an enum class from entry initializers of this enum
+
+> **Issue**: [KT-49110](https://youtrack.jetbrains.com/issue/KT-49110)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.9 will prohibit all kinds of access to the companion object of an enum from an enum entry initializer
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: report a warning (or an error in the progressive mode) on such companion member access
+> - 1.9.0: raise the warning to an error,
+>   `-XXLanguage:-ProhibitAccessToEnumCompanionMembersInEnumConstructorCall` can be used to temporarily revert to the pre-1.8 behavior
+
+### Deprecate and remove `Enum.declaringClass` synthetic property
+
+> **Issue**: [KT-49653](https://youtrack.jetbrains.com/issue/KT-49653)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin allowed using the synthetic property `declaringClass` on `Enum` values produced from
+> the method `getDeclaringClass()` of the underlying Java class `java.lang.Enum` even though this method is not available
+> for Kotlin `Enum` type. Kotlin 1.9 will prohibit using this property, proposing to migrate to the extension property
+> `declaringJavaClass` instead
+>
+> **Deprecation cycle**:
+>
+> - 1.7.0: report a warning (or an error in the progressive mode) on `declaringClass` property usages,
+>   propose the migration to `declaringJavaClass` extension
+> - 1.9.0: raise the warning to an error,
+>   `-XXLanguage:-ProhibitEnumDeclaringClass` can be used to temporarily revert to the pre-1.9 behavior
+> - \>=1.10: remove `declaringClass` synthetic property
+
+### Deprecate the `enabled` and the `compatibility` modes of the compiler option -Xjvm-default
```

Comment:
```suggestion
### Deprecate the `enable` and the `compatibility` modes of the compiler option -Xjvm-default
```

Reply:
It's called "enable" https://github.com/JetBrains/kotlin/blob/93fb7597e912e3d356fe12dbbdb4159d958f234e/compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/arguments/K2JVMCompilerArguments.kt#L315

Reply:
Good catch, makes sense to fix in the previous version compatibility guide.

Reply:
I fixed it in old guides, but it was so easy to miss this resolved comment!😅